### DISCUSSION TITLE
Dropping support for older frameworks in the WireMock.Net NuGet

### DISCUSSION THREAD

**Discussion by StefH** [wiremock/WireMock.Net#1060] [Polls]
I want to drop support for older frameworks in the [WireMock.Net NuGet package](https://www.nuget.org/packages/WireMock.Net).

### netstandard 1.3
I want to drop support for netstandard 1.3 because:
- The WireMock.Net NuGet package is always used in the context of a unit-test project, and I think that a unit-test project should always target .NET (core) or .NET Full Framework.

### netstandard 2.0 & netstandard 2.1
For these two, the same context applies as for 1.3

---

## :two: .NET Framework 4.5.1, 4.6 and 4.6.1
All these full frameworks have gone end of life (https://github.com/WireMock-Net/WireMock.Net/issues/864) so these could be removed. (However if someone is still using these, they need to upgrade)

---

## :three: .NET Core 3.1 and .NET 5.0
These full frameworks have also reached end of life.

---

## :bulb: Versioning
If this change is applied to WireMock.Net, this will mean a version change from `1.x` to `2.x`.

**Comment by StefH** [comment]
Tagging...

@Netizine
@cezarypiatek
@mcopjan
@jzabroski 
@kharabasz 
@sergioprates 
@robertmircea
@MindaugasLaganeckas 
@MJLHThomassenHadrian
@mark-hingston
@basdijkstra 
@jflevesque-genetec
@sameena-ops 
@leolplex
@alastairtree
@NoahLerner
@phillee007
@vitaliydavydiak
@akamud
@kashifsoofi
@ggradnig
@eli-darkly
@cal-schleupen
@oleg-nenashev

**Comment by basdijkstra** [comment]
I voted for dropping _all_ framework versions that are no longer officially supported, but that is mostly speaking as a fellow open source library maintainer rather than a user.

Reasons:
* Users can always stick to 1.5.x if they are in older versions
* It’ll make your life much easier moving forward and implementing new features
* Less important, but at some point you’ll have to anyway

**Comment by jzabroski** [comment]
I may have accidentally voted. Just literally woke up and tapped my thumb.

Agree, supporting old  .net core version is insane.

Can you target net482 tho? Not a huge deal if you can't - just add a note on the Readme for people who want that version to use the last supported nuget package and say any issues are unsupported

**Reply by StefH** [reply]
.NET Framework 4.8.2 does not yet exists (https://en.wikipedia.org/wiki/.NET_Framework_version_history), did you mean 4.7.2 ?

**Reply by jzabroski** [reply]
Meant 4.8.1.

**Comment by StefH** [comment]
![image](https://github.com/WireMock-Net/WireMock.Net/assets/249938/58ec582a-6418-4ad7-a8e7-1ea24f5cd0c1)

**Reply by jzabroski** [reply]
netstandard works - it should support net481 in that case?

**Comment by NoahLerner** [comment]
Regarding versioning, don't you think it would be appropriate to jump to 2.x.x?
Personally, I like semantic versioning & dropping support for EOL frameworks seems to be a significant breaking change.

**Reply by StefH** [reply]
Indeed, thanks for the suggestion. When I will implement this, a version change to 2.x sounds good.

**Comment by LevYas** [comment]
I agree with dropping all the old frameworks, your comfort with development is of the highest priority.
Jumping through hoops to workaround some compatibility issues and overly complex build configurations leads to frustration and burnout quite fast, we don't need that.

**Comment by StefH** [comment]
Note that I'll pick this up in 2025
